pkg/dump/skip_defaults_test.go

```diff
@@ -0,0 +1,489 @@
+			expected:     true,
+		},
+		{
+			name:         "int slices",
+			fieldSlice:   []int{1, 2, 3},
+			defaultSlice: []int{1, 2, 3},
+			expected:     true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fieldVal := reflect.ValueOf(tt.fieldSlice)
+			defaultVal := reflect.ValueOf(tt.defaultSlice)
+			result := compareSlices(fieldVal, defaultVal)
+			if result != tt.expected {
+				t.Errorf("compareSlices(%v, %v) = %v, expected %v", tt.fieldSlice, tt.defaultSlice, result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestCompareMaps(t *testing.T) {
+	tests := []struct {
+		name       string
+		fieldMap   map[string]interface{}
+		defaultMap map[string]interface{}
+		expected   map[string]interface{}
+	}{
+		{
+			name: "remove default values",
+			fieldMap: map[string]interface{}{
+				"timeout": 5000,
+				"retries": 3,
+				"name":    "test",
+			},
+			defaultMap: map[string]interface{}{
+				"timeout": 5000,
+				"name":    "default",
+			},
+			expected: map[string]interface{}{
+				"retries": 3,
+				"name":    "test",
+			},
+		},
+		{
+			name: "nested maps",
+			fieldMap: map[string]interface{}{
+				"config": map[string]interface{}{
+					"timeout":   5000,
+					"retries":   3,
+					"protocols": []string{"http", "https"},
+				},
+				"enabled": true,
+				"version": "1.0",
+			},
+			defaultMap: map[string]interface{}{
+				"config": map[string]interface{}{
+					"timeout":   5000,
+					"protocols": []string{"http", "https"},
+				},
+				"enabled": false,
+				"version": "1.0",
+			},
+			expected: map[string]interface{}{
+				"config": map[string]interface{}{
+					"retries": 3,
+				},
+				"enabled": true,
+			},
+		},
+		{
+			name: "all values are defaults",
+			fieldMap: map[string]interface{}{
+				"enabled": false,
+				"version": "1.0",
+				"config": map[string]interface{}{
+					"timeout":   5000,
+					"protocols": []string{"http", "https"},
+				},
+			},
+			defaultMap: map[string]interface{}{
+				"enabled": false,
+				"version": "1.0",
+				"config": map[string]interface{}{
+					"timeout":   5000,
+					"protocols": []string{"http", "https"},
+				},
+			},
+			expected: map[string]interface{}{},
+		},
+		{
+			name: "no values are defaults",
+			fieldMap: map[string]interface{}{
+				"enabled": true,
+				"version": "2.0",
+				"config": map[string]interface{}{
+					"timeout":   6000,
+					"protocols": []string{"grpc", "https"},
+				},
+			},
+			defaultMap: map[string]interface{}{
+				"enabled": false,
+				"version": "1.0",
+				"config": map[string]interface{}{
+					"timeout":   5000,
+					"protocols": []string{"http", "https"},
+				},
+			},
+			expected: map[string]interface{}{
+				"enabled": true,
+				"version": "2.0",
+				"config": map[string]interface{}{
+					"timeout":   6000,
+					"protocols": []string{"grpc", "https"},
+				},
+			},
+		},
+		{
+			name: "field has nil value",
+			fieldMap: map[string]interface{}{
+				"timeout": 5000,
+				"name":    nil,
```

Comment:
If `name` has a non-null default - and user has configured it to be null - does it still get removed?